azsettings/cloud_settings.go

```diff
@@ -0,0 +1,96 @@
+package azsettings
+
+import (
+	"fmt"
+	"sort"
+)
+
+type AzureCloudInfo struct {
+	Name        string
+	DisplayName string
+}
+
+type AzureCloudSettings struct {
+	Name         string
+	DisplayName  string
+	AadAuthority string
+	Properties   map[string]string
+}
+
+var predefinedClouds = map[string]*AzureCloudSettings{
+	AzurePublic: {
+		Name:         "AzureCloud",
+		DisplayName:  "Azure",
+		AadAuthority: "https://login.microsoftonline.com/",
+		Properties: map[string]string{
+			"azureDataExplorerSuffix":   ".kusto.windows.net",
+			"logAnalytics":              "https://api.loganalytics.io",
+			"logAnalyticsResourceId":    "https://api.loganalytics.io",
```

Comment:
Why do we need `logAnalyticsResourceId` and `resourceManagerResourceId` when they're duplicates of the fields above?

Reply:
Removed redundant properties.